src/redux/reducers/priceDataReducers.ts

```diff
@@ -0,0 +1,28 @@
+import { FETCH_PRICE_DATA, PRICE_DATA_LOADED } from '../actions/priceDataActions';
+import { APIStatusEnum, IPriceDataAction, IPriceDataState } from '../ApiInterfaces';
+
+const initialState: IPriceDataState = {
+    status: APIStatusEnum.IDLE,
+    data: [],
+};
+
+const priceData = (state = initialState, action: IPriceDataAction): IPriceDataState => {
```

Comment:
you should conside moving to redux toolkit, it is amazing, you have a small number of reducers that you can rewrite in a 10-20 min
RTK offers a lot of typescript support for actions